### ISSUE TITLE
flutter_test depends on analyzer 0.33.0-alpha.0;  json_serializable depends on analyzer ^0.32.2; things are bad

### ISSUE THREAD

**Issue by cikichen** [dart-lang/json_serializable#339]
Because every version of flutter_test from sdk depends on analyzer 0.33.0-alpha.0 and json_serializable >=1.0.0 depends on analyzer ^0.32.2, flutter_test from sdk is incompatible with json_serializable >=1.0.0.
So, because community depends on both json_serializable ^1.5.0 and flutter_test any from sdk, version solving failed.

pub get failed (1)

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v0.9.7-pre.61, on Mac OS X 10.14 18A391, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK 28.0.2)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
[✓] Android Studio (version 3.2)
[✓] VS Code (version 1.27.2)
[✓] Connected device (1 available)
```

**Comment by kevmoo** [comment]
You may have to put a dependency override in for analyzer in the short run. I'm asking around internally...

**Comment by kevmoo** [comment]
Fixed in https://github.com/dart-lang/json_serializable/commit/ce27bceb34442dcaad6645d904497da47b2304a2 – published as 1.5.1
